persist/log.go

```diff
@@ -119,6 +125,22 @@ func (cf *closeableFile) Write(b []byte) (int, error) {
 // NewFileLogger returns a logger that logs to logFilename. The file is opened
 // in append mode, and created if it does not exist.
 func NewFileLogger(logFilename string) (*Logger, error) {
+	// if a global log directory has been specified, strip the base path and use the explicit LogDir
+	// TODO: use a less naïve path validator here. Currently, a non-existent path will result in failure.
+	// This will mean we need to:
+	//   * validate input to create path
+	//   * check if directory exists, else attempt to create
+	//     - this seems to exist already for profile directories and the like,
+	//       is there file utility code in this repo I can rely on?
+	//   * once the above is done, don't overrwrite logFilename if these conditions
+	//     fail, and instead just write to the standard log path.
+	//     - it would be ideal if we could return an error here, noting the fallback,
+	//       however none of the callers of this function are prepared for that type
+	//       of error, meaning we need to decide to either modify the callers, or just
+	//       call an fmt.Println()
```

Comment:
`LogDir` should be created in the startup code (see other comment), so you don't need to create it in `NewFileLogger` or implement any fallback code. We'd definitely prefer to fail with an error here instead of writing to an unintended path.